mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,487 @@
+
+    def _log_metrics(self):
+        """
+        Helper method to log metrics into specified run.
+        """
+        timestamp = int(time.time() * 1000)
+        self.client.log_batch(
+            self.run_id,
+            metrics=[
+                Metric(key=_gen_log_key(key, self.dataset_name),
+                       value=value, timestamp=timestamp, step=0)
+                for key, value in self.metrics.items()
+            ],
+        )
+
+    def _log_image_artifact(
+        self, do_plot, artifact_name,
+    ):
+        import matplotlib.pyplot as pyplot
+        artifact_file_name = _gen_log_key(artifact_name, self.dataset_name) + '.png'
+        artifact_file_local_path = self.temp_dir.path(artifact_file_name)
+
+        try:
+            pyplot.clf()
+            do_plot()
+            pyplot.savefig(artifact_file_local_path)
+        finally:
+            pyplot.close(pyplot.gcf())
+
+        mlflow.log_artifact(artifact_file_local_path)
+        artifact = ImageEvaluationArtifact(uri=mlflow.get_artifact_uri(artifact_file_name))
+        artifact.load(artifact_file_local_path)
+        self.artifacts[artifact_name] = artifact
+
+    def _log_pandas_df_artifact(
+        self, pandas_df, artifact_name
+    ):
+        artifact_file_name = _gen_log_key(artifact_name, self.dataset_name) + '.csv'
+        artifact_file_local_path = self.temp_dir.path(artifact_file_name)
+        pandas_df.to_csv(artifact_file_local_path, index=False)
+        mlflow.log_artifact(artifact_file_local_path)
+        artifact = CsvEvaluationArtifact(
+            uri=mlflow.get_artifact_uri(artifact_file_name),
+            content=pandas_df,
+        )
+        artifact.load(artifact_file_local_path)
+        self.artifacts[artifact_name] = artifact
+
+    def _log_model_explainability(self):
+        if not self.evaluator_config.get('log_model_explainability', True):
+            return
+
+        if self.model_loader_module == 'mlflow.spark':
+            # TODO: Shap explainer need to manipulate on each feature values,
+            #  but spark model input dataframe contains Vector type feature column
+            #  which shap explainer does not support.
+            #  To support this, we need expand the Vector type feature column into
+            #  multiple scaler feature columns and pass it to shap explainer.
+            _logger.warning(
+                'Log model explainability currently does not support spark model.'
+            )
+            return
+
+        try:
+            import shap
+            import matplotlib.pyplot as pyplot
+        except ImportError:
+            _logger.warning(
+                'Shap or matplotlib package is not installed, Skip log model explainability.'
+            )
+
+        if Version(shap.__version__) < Version('0.40'):
+            _logger.warning(
+                'Shap package version is lower than 0.40, Skip log model explainability.'
```

Comment:
Nice check! :). Can we document this in the API docs for `mlflow.evaluate()`?

```suggestion
                'SHAP package version is lower than 0.40, so model explainability insights will not be logged.'
```